### DISCUSSION TITLE
How to search or filter by document id?

### DISCUSSION THREAD

**Discussion by Amtech281** [firecmsco/firecms#315] [Q&A]
Hi
a question: Having a collection of several thousands documents we would like to filter by ID, however this is not supported as it seems. Is there a way to filter by the document id?
thanks

**Comment by fgatti675** [comment]
Hi Andre, right now the only way is if you have text search enabled, you index the id in Algolia and then you type the id in the search text box in the collection

**Comment by fgatti675** [comment]
Alternatively, you can open any entity and just modify the URL, probably easier!

**Reply by Amtech281** [reply]
hi Francesco, thanks for the feedback. Your recommendation with modify the URL I did not understand, what URL?

**Comment by Benelisha** [comment]
https://github.com/Benelisha/FireCMS-free-search
I added simple search if you want, hope it helps. let me know if you have any issues, its not the best but it works and no need for third party paid services.
